config/puma.rb

```diff
@@ -24,7 +24,13 @@
 if rails_env == "production"
   on_worker_boot do
     require "active_record"
-    ActiveRecord::Base.connection.disconnect! rescue ActiveRecord::ConnectionNotEstablished
-    ActiveRecord::Base.establish_connection(YAML.load_file("#{app_path}/config/database.yml")[rails_env]["primary"])
+    begin
+      ActiveRecord::Base.connection.disconnect!
+    rescue StandardError
+      ActiveRecord::ConnectionNotEstablished
+    end
+    ActiveRecord::Base.establish_connection(
+      YAML.load_file("#{app_path}/config/database.yml")[rails_env]["primary"]
+    )
   end
 end
```

Comment:
@mo-nathan I can't figure out these test failures. CI is getting "invalid authenticity token".

These tests do not fail locally and they're only failing on CI "push" - note that CI "pull request" passes. They are mostly not system tests and so they don't seem related to solid queue and there are no changes in test environment configs.

There is a change however here in `config.puma.rb`, to run puma as the test server. This was necessary to get the system tests to use Puma (and thereby have jobs/cable avaliable), otherwise the server wasn't set up on port 3000. (See also below in `test/application_system_test_case.rb:28` `Capybara.server = :puma`)

Reply:
Fixed it. It just needed to not have the port set, in test environment.